src/bits/mod.rs

```diff
@@ -1,17 +1,14 @@
-use crate::secret_sharing::BooleanShare;
-
+use crate::secret_sharing::SharedValue;
 use generic_array::{ArrayLength, GenericArray};
-
 use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Index, Not};
 
-mod bit_array;
-
-pub use bit_array::{BitArray40, BitArray8};
+mod shared_value_array;
+pub use shared_value_array::{BitArray40, BitArray8};
 
 /// Trait for data types storing arbitrary number of bits.
 // TODO: Implement `Message`
-pub trait BitArray:
-    BooleanShare + TryFrom<u128> + Into<u128> + Index<usize, Output = bool> + Index<u32, Output = bool>
+pub trait SharedValueArray:
```

Comment:
This is maybe a bit misleading given the implementation of these.  It might be better to think of this as `Fp2Array` instead.

Reply:
Renamed to `Fp2Array`.